### PR TITLE
Add config for data index url

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by mswiderski** [comment]
@cristianonicolai doesn't this tights us to vert.x web layer and in case we would like to go with undertow then this won't work? just thinking if there is any other solution possible....

**Comment by cristianonicolai** [comment]
@mswiderski I think it does, as in it is probably not possible to support multiple web layers in Quarkus. This being a relatively small fix if we decide to use something else, should be quite simple to migrate.

File: 04-kogito-travel-agency/src/main/java/org/kie/kogito/app/VertxRouter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @ConfigProperty(name = "kogito.dataindex.url", defaultValue = "http://localhost:8180/graphql")
+    private String dataIndexURL;
+
+    @Route(path = "/", methods = GET)
+    public void handle(RoutingContext context) {
+        try {
+            String resource = context.vertx().fileSystem()
+                    .readFileBlocking("META-INF/resources/index.html")
+                    .toString(UTF_8)
+                    .replace("__GRAPHIQL_ENDPOINT__", "\"" + dataIndexURL + "\"");
```
Comment (mswiderski) [inline 04-kogito-travel-agency/src/main/java/org/kie/kogito/app/VertxRouter.java]: just wondering... should we do this only once?

Comment (mswiderski) [reply 04-kogito-travel-agency/src/main/java/org/kie/kogito/app/VertxRouter.java]: I mean the reading and replacing...

Comment (cristianonicolai) [reply 04-kogito-travel-agency/src/main/java/org/kie/kogito/app/VertxRouter.java]: true, Improved it a bit now.

**Review by mswiderski — APPROVED** [review]

**Pull request merged**